src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,40 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private final BulldozerProducer bulldozer = new BulldozerProducer();
+    private final ExcavatorProducer excavator = new ExcavatorProducer();
+    private final TruckProducer truck = new TruckProducer();
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines = new ArrayList<>();
+        if (type.equals(Bulldozer.class)) {
+            return (List<Machine>) (machines = bulldozer.get());
+        } else if (type.equals(Excavator.class)) {
+            return (List<Machine>) (machines = excavator.get());
+        } else if (type.equals(Truck.class)) {
+            return (List<Machine>) (machines = truck.get());
+        } else {
+            machines = new ArrayList<>();
+        }
```

Comment:
`machines` variable useless. just do returns.
And remove `else { machines = new ArrayList<>();  }` better to return `Collection.emptyList()` in the end